spec/http.md

```diff
@@ -0,0 +1,51 @@
+HTTP
+====
+
+## Resources
+
+Each EC2 instance MUST present the following (GET) resources:
+
+    [/service/healthcheck](#healthcheck)
+    [/service/gtg](#gtg)
+    [/service/dependencies](#dependencies)
+
+These resources should be presented on port 8080 (not port 80).
+
+### <a name="healthcheck">Healthcheck</a>
+
+Is the service running? Two possible responses:
+
+    200 "OK" (service is running)
+    50* (service is down - choose exact status code as appropriate)
+
+### <a name="gtg">GTG</a>
+
+    200 "OK" (service is able to usefully respond to requests)
+    50* (service is down - choose exact status code as appropriate)
+
+Is the service able to usefully respond to requests? (Typically, this
+indicates whether the service is running and its dependencies are
+available as expected.)
```

Comment:
The problem with having a single healthcheck that only returns 200 OK when its dependencies are healthy is that if one of your dependencies fail the autoscaling group will keep terminating/launching instances unnecessarily.